.github/workflows/build-minio-image.yaml

```diff
@@ -0,0 +1,108 @@
+  workflow_dispatch:
+    inputs:
+      minio_version:
+        description: 'MinIO version (e.g., RELEASE.2025-01-15T12-00-00Z) or leave empty for latest'
+        required: false
+        default: ''
+
+jobs:
+  build-and-push-minio:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v5
+
+      - name: Detect latest MinIO version
+        id: version
+        run: |
+          # Use manual input if provided, otherwise fetch from GitHub API
+          if [ -n "${{ github.event.inputs.minio_version }}" ]; then
+            VERSION="${{ github.event.inputs.minio_version }}"
+            echo "Using manually specified version: $VERSION"
+          else
+            echo "Fetching latest MinIO release from GitHub API..."
+            VERSION=$(curl -s https://api.github.com/repos/minio/minio/releases/latest | \
+              grep '"tag_name":' | \
+              sed -E 's/.*"([^"]+)".*/\1/')
+            echo "Detected latest version: $VERSION"
+          fi
+
+          echo "minio_version=$VERSION" >> "$GITHUB_OUTPUT"
+
+      - name: Check if image already exists
+        id: check
+        run: |
+          echo "Checking if kurlsh/minio:${{ steps.version.outputs.minio_version }} exists..."
+
+          if docker manifest inspect kurlsh/minio:${{ steps.version.outputs.minio_version }} > /dev/null 2>&1; then
+            echo "exists=true" >> "$GITHUB_OUTPUT"
+            echo "Image already exists. Skipping build."
+          else
+            echo "exists=false" >> "$GITHUB_OUTPUT"
+            echo "Image does not exist. Proceeding with build."
+          fi
+
+      - name: Set up Docker Buildx
+        if: steps.check.outputs.exists == 'false'
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to Docker Hub
+        if: steps.check.outputs.exists == 'false'
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USER }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      - name: Build and push MinIO image
+        if: steps.check.outputs.exists == 'false'
+        uses: docker/build-push-action@v6
+        with:
+          context: ./addons/minio/build-images/minio
+          file: ./addons/minio/build-images/minio/Dockerfile
+          build-args: |
+            RELEASE=${{ steps.version.outputs.minio_version }}
+          push: true
+          tags: |
+            kurlsh/minio:${{ steps.version.outputs.minio_version }}
+            kurlsh/minio:latest
+          platforms: linux/amd64
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Run Trivy vulnerability scanner
+        if: steps.check.outputs.exists == 'false'
+        uses: aquasecurity/trivy-action@0.30.0
+        with:
+          image-ref: kurlsh/minio:${{ steps.version.outputs.minio_version }}
+          format: 'sarif'
+          output: 'trivy-results.sarif'
+          severity: 'HIGH,CRITICAL'
+          exit-code: '1'
```

Comment:
Will this prevent uploading results when there is a cve?

Reply:
it should not as we are using `always()`  condition for the upload 

https://github.com/replicatedhq/kURL/blob/3be0f1b56310d5813bb98b1895cf36687b82be31/.github/workflows/build-minio-image.yaml#L89